tests/atoms/gaia_data_layer.js

```diff
@@ -207,27 +201,23 @@ var GaiaDataLayer = {
   },
 
   removeContact: function(aContact, aCallback) {
+    // requires the 'contacts-write' persmission
     var callback = aCallback || marionetteScriptFinished;
-    SpecialPowers.addPermission('contacts-write', true, document);
     console.log('removing contact with id \'' + aContact.id + '\'');
     var req = window.navigator.mozContacts.remove(aContact);
     req.onsuccess = function() {
       console.log('success removing contact with id \'' + aContact.id + '\'');
-      SpecialPowers.removePermission('contacts-write', document);
       callback(true);
     };
     req.onerror = function() {
       console.error('error removing contact with id \'' +
                       aContacts[i].id + '\'');
-      SpecialPowers.removePermission('contacts-write', document);
       callback(false);
     };
   },
 
   getSetting: function(aName, aCallback) {
     var callback = aCallback || marionetteScriptFinished;
-    SpecialPowers.addPermission('settings-read', true, document);
-    SpecialPowers.addPermission('settings-api-read', true, document);
```

Comment:
Nit: Could you add a comment at the top of the function stating that these permissions are required?